x-pack/plugins/apm/public/components/app/service_dashboards/actions/save_dashboard_modal.tsx

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useState } from 'react';
+import {
+  EuiButton,
+  EuiModal,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSwitch,
+  EuiModalBody,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiFlexGroup,
+  EuiToolTip,
+  EuiIcon,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { DashboardItem } from '@kbn/dashboard-plugin/common/content_management';
+import { callApmApi } from '../../../../services/rest/create_call_apm_api';
+import { useDashboardFetcher } from '../../../../hooks/use_dashboards_fetcher';
+import { FETCH_STATUS } from '../../../../hooks/use_fetcher';
+import { useApmPluginContext } from '../../../../context/apm_plugin/use_apm_plugin_context';
+import { useApmParams } from '../../../../hooks/use_apm_params';
+import { SERVICE_NAME } from '../../../../../common/es_fields/apm';
+import { MergedServiceDashboard } from '..';
+
+interface Props {
+  onClose: () => void;
+  onRefresh: () => void;
+  currentDashboard?: MergedServiceDashboard;
+  serviceDashboards?: MergedServiceDashboard[];
+}
+
+export function SaveDashboardModal({
+  onClose,
+  onRefresh,
+  currentDashboard,
+  serviceDashboards,
+}: Props) {
+  const {
+    core: { notifications },
+  } = useApmPluginContext();
+  const { data: allAvailableDashboards, status } = useDashboardFetcher();
```

Comment:
You are not passing in a `query`. Does that mean you are loading all the dashboards? We should only load the `title` of the linked dashboards.

Reply:
as we discussed offline, we need to request the the dashboard for the selector dropdown. I created a ticket to lazy load the options. https://github.com/elastic/kibana/issues/167761